TypeCobol.LanguageServer/TypeCobolCustomLanguageServer/TypeCobolCustomLanguageServer.cs

```diff
@@ -23,29 +24,35 @@ public TypeCobolCustomLanguageServer(IRPCServer rpcServer, Queue<MessageActionWr
         /// </summary>
         public bool UseOutlineRefresh { get; set; }
 
-        protected override void OnShutdown()
-        {
-            if (UseOutlineRefresh && this.Workspace.DocumentModifiedEvent != null)
-                this.Workspace.DocumentModifiedEvent -= DocumentModified;
+        /// <summary>
+        /// Are we using the CFG view in the client.    
+        /// </summary>
+        public bool UseCfgDfaDataRefresh { get; set; }
 
-            base.OnShutdown();
+        protected override InitializeResult OnInitialize(InitializeParams parameters)
+        {
+            var result = base.OnInitialize(parameters);
+            this.Workspace.DocumentModifiedEvent += DocumentModified;
+            this.Workspace.ClientConfigurationChangedEvent += OnClientOptionsChanged;
+            return result;
         }
 
-        protected override void OnDidOpenTextDocument(DidOpenTextDocumentParams parameters)
+        /// <summary>
+        /// Handle -ol and -cfg options from the client configuration change notification.
+        /// </summary>
+        /// <param name="sender">Should be the Workspace instance</param>
+        /// <param name="options">Client's Options</param>
+        private void OnClientOptionsChanged(object sender, IEnumerable<string> options)
```

Comment:
Yes it is for properties that come from the Client plugin an that are not TypeCobol.CLI properties, like -ol (Outline) these are TypeCobol Custom and TypeCobolCustomLanguageServer set some flags depending on it.

Reply:
I don't think we need that. In `TypeCobolCustomLanguageServer` class, you can override `OnDidChangeConfiguration` method and do the same job. So no need for this event to get config data back from the Workspace.

Reply:
It is necessary because -ol and -cfg options also come from the command line, and are set via TypeCobolCustomLanguageServer when no Workspace is instantiate yet. So it is TypeCobolCustomLanguageServer which host UseOutlineRefresh  and UseCfgDfaDataRefresh.

Reply:
But you are right for -ol and -cfg options inside LSRTest, I am going to parameterized these options in the custom config.json generated.